pkg/reconciler/sequence/sequence.go

```diff
@@ -333,3 +340,147 @@ func (r *Reconciler) removeUnwantedSubscriptions(ctx context.Context, seq *v1.Se
 
 	return nil
 }
+
+func (r *Reconciler) reconcileEventPolicies(ctx context.Context, s *v1.Sequence, channels []*eventingduckv1.Channelable, subs []*messagingv1.Subscription, featureFlags feature.Flags) error {
+	if !featureFlags.IsOIDCAuthentication() {
+		return r.cleanupAllEventPolicies(ctx, s)
+	}
+
+	existingPolicies, err := r.listEventPoliciesForSequence(s)
+	if err != nil {
+		return fmt.Errorf("failed to list existing EventPolicies: %w", err)
+	}
+
+	// Prepare maps for efficient lookups, updates, and deletions of policies
+	existingPolicyMap := make(map[string]*eventingv1alpha1.EventPolicy)
+	for _, policy := range existingPolicies {
+		existingPolicyMap[policy.Name] = policy
+	}
+
+	// Prepare lists for different actions so that policies can be categorized
+	var policiesToUpdate, policiesToCreate []*eventingv1alpha1.EventPolicy
+	policiesToDelete := make([]*eventingv1alpha1.EventPolicy, 0, len(existingPolicies))
+
+	// Handle intermediate channel policies (skip the first channel as it's the input channel!)
+	for i := 1; i < len(channels); i++ {
+		expectedPolicy := resources.MakeEventPolicyForSequenceChannel(s, channels[i], subs[i-1])
+		existingPolicy, exists := existingPolicyMap[expectedPolicy.Name]
+
+		if exists {
+			if !equality.Semantic.DeepDerivative(expectedPolicy, existingPolicy) {
+				policiesToUpdate = append(policiesToUpdate, expectedPolicy)
```

Comment:
I am seeing 
```
{"level":"info","ts":"2024-08-07T06:09:42.541Z","logger":"controller.event-broadcaster","caller":"record/event.go:364","msg":"Event(v1.ObjectReference{Kind:\"Sequence\", Namespace:\"default\", Name:\"my-sequence\", UID:\"b70f7844-6bfb-4336-9805-9da269e9270e\", APIVersion:\"flows.knative.dev/v1\", ResourceVersion:\"3815\", FieldPath:\"\"}): type: 'Warning' reason: 'InternalError' failed to reconcile EventPolicies: failed to update EventPolicies: eventpolicies.eventing.knative.dev \"my-sequence-my-sequence-kn-sequence-1\" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update","commit":"5da98e0-dirty","knative.dev/pod":"eventing-controller-7ccd48bcb-jlgsz"}
```
in the controller. So I guess you need to set the resource version on `expectedPolicy` before doing the update. So something like
```
expectedPolicy.SetResourceVersion(existingPolicy.ResourceVersion)
policiesToUpdate = append(policiesToUpdate, expectedPolicy)
...
```

Reply:
Can we also add a test for this? For example have the "intermediate" event policies already existing (those which you create for the input channel and channels of the steps), but requiring an update (e.g. by having one of those policies having a "wrong" spec). And then expect and update on that policy by the reconciler.

Same probably makes sense to test, that policies get cleaned up / deleted.